### PR TITLE
Bump go-ethereum and go version

### PR DESCRIPTION
**Pull request by avichalp**
# Summary

This PR updates `go-ethereum` version to `v11.3` and updates golang version to `v1.19`, and tests the validator on go `v1.20`. 

# Context

[go-ethereum](https://github.com/ethereum/go-ethereum/releases/tag/v1.11.3) has the policy to support only the last two versions of golang. It means that after go `1.20` release, go `1.18` will not be supported by the latest version of the `go-ethereum`. In this PR, we bump the version of the following: 

- go-ethereum from `10.26` to `11.3`
- golang from `1.18` to `1.19` and `1.20`

We want to run the tests on go `1.20` to make the update process smoother in the future. If we face an issue with go 1.20, we will be alerted immediately.

# Implementation overview

- We bump the `go-ethereum` version to 11.3. 
- We build the binaries using go 1.19 and run the tests on both go 1.19 and 1.20 

# Implementation details and review orientation

We have updated the test workflow file for running the tests using both go 1.19 and 1.20. For building using go 1.19 we have updated the docker files. To update the go-ethereum version, we have bumped its version in `go.mod` and changed the go version to 1.19 too.

# Checklist

- [x]  Are changes backward compatible with existing SDKs, or is there a plan to manage it correctly?
- [x]  Are changes covered by existing tests, or were new tests included?
- [x]  Are code changes optimized for future code readers, commenting on problematic areas to understand (if any)?
- [x]  Future-self question: Did you avoid unjustified/unnecessary complexity to achieve the goal?

### CODE REVIEW

File: cmd/api/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19.5-alpine as builder
+FROM golang:1.19.6-alpine as builder
```
Comment (avichalp) [inline cmd/api/Dockerfile:1]: We are already using `1.19.5` in the builder, we PR will update it to `1.19.6`.

File: go.mod [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-go 1.18
+go 1.19
 
 require (
 	cloud.google.com/go/bigquery v1.45.0
 	cloud.google.com/go/logging v1.6.1
 	github.com/XSAM/otelsql v0.20.0
-	github.com/ethereum/go-ethereum v1.10.26
+	github.com/ethereum/go-ethereum v1.11.3
```
Comment (avichalp) [inline go.mod:9]: go-ethereum v1.11.3 drops the support for go v1.18; hence we are updating to go 1.19.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -14,11 +14,11 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.18.x, 1.19.x]
+        go-version: [1.19.x, 1.20.x]
```
Comment (avichalp) [inline .github/workflows/test.yml:17]: Tests will run on 1.19 and 1.20 like before with 1.18 and 1.19.

**Review by brunocalza — DISMISSED** [review]
Looks good!

**Review by brunocalza — APPROVED** [review]

**Pull request merged**